server/src-lib/Hasura/RQL/DDL/Schema.hs

```diff
@@ -58,18 +58,20 @@ data RunSQL
   { rSql                      :: Text
   , rCascade                  :: !(Maybe Bool)
   , rCheckMetadataConsistency :: !(Maybe Bool)
+  , rReadOnly                 :: !(Maybe Bool)
   } deriving (Show, Eq, Lift)
 $(deriveJSON (aesonDrop 1 snakeCase){omitNothingFields=True} ''RunSQL)
 
 runRunSQL :: (CacheBuildM m, UserInfoM m) => RunSQL -> m EncJSON
-runRunSQL (RunSQL t cascade mChkMDCnstcy) = do
+runRunSQL (RunSQL t cascade mChkMDCnstcy mReadOnly) = do
   adminOnly
+  let txAccessMode = fmap boolToAccessMode mReadOnly
```

Comment:
If `read_only` mode is `true` we shouldn't check for `isAltDropReplace`. AFAIK, Postgres doesn't allow to execute DDL statements in `READY ONLY` mode transaction. It is safe to directly execute SQL without metadata check. Postgres will throw exceptions if the query contains any DDL statements.

`withMetadataCheck` function shouldn't account for `txAccessMode` :wink: . It is always `Q.ReadWrite`.

Reply:
Oh yeah. I didn't want too much diff here so haven't touched this bit.